docs/ssl-input.asciidoc

```diff
@@ -0,0 +1,71 @@
+SSL/TLS is disabled by default. Besides enabling it, you need to provide a certificate and a corresponding
+private key as well.
+
+Following is a basic APM Server SSL config with secure communication enabled:
+
+[source,yaml]
+----
+apm-server.ssl.enabled: true
+apm-server.ssl.certificate: "/path/to/apm-server.crt"
+apm-server.ssl.key: "/path/to/apm-server.key"
+----
+
+This will make APM Server serve HTTPS requests instead of HTTP.
+
+A full list of configuration options is available in <<agent-server-ssl>>.
+
+Certificate and private key can be issued by a trusted certificate authority (CA)
+or <<self-signed-cert,self-signed>>.
+
+NOTE: When using a self-signed (or custom CA) certificate, communication from APM Agents will require
+additional settings due to <<ssl-server-authentication>>
+
+[[self-signed-cert]]
+==== Creating a self-signed certificate
+
+The Elasticsearch distribution offers the `certutil` tool for the creation of self-signed certificates:
+
+1. Create a CA: `./bin/elasticsearch-certutil ca --pem`. You'll be prompted to enter the desired
+location of the output zip archive containing the certificate and the private key.
+2. Extract the contents of the CA archive.
+3. Create the self-signed certificate: `./bin/elasticsearch-certutil cert --ca-cert
+<path-to-ca-crt>/ca.crt --ca-key <path-to-ca-key>/ca.key --pem --name localhost`
+4. Extract the certificate and key from the resulted zip archive.
+
+[[ssl-server-authentication]]
+==== Server certificate authentication
+
+By default, when SSL is enabled for APM Server inbound communication, agents will verify the identity
+of the APM Server by authenticating its certificate. It is not recommended to change this default,
+however it is possible through agent configuration:
+
+* *Go Agent*: {apm-go-ref}/configuration.html#config-verify-server-cert[`ELASTIC_APM_VERIFY_SERVER_CERT`]
+* *.NET Agent*: {apm-dotnet-ref}/config-reporter.html#config-verify-server-cert[`VerifyServerCert`]
+* *Java Agent*: {apm-java-ref}/config-reporter.html#config-verify-server-cert[`verify_server_cert`]
+* *Python Agent*: {apm-py-ref}/configuration.html#config-verify-server-cert[`verify_server_cert`]
+* *Ruby Agent*: {apm-ruby-ref}/configuration.html#config-verify-server-cert[`verify_server_cert`]
+* *NodeJS Agent*: {apm-node-ref}/configuration.html#validate-server-cert[`verifyServerCert`]
+
+When the APM server uses a certificate that is not chained to a publicly-trusted certificate
+(e.g. self-signed), additional setting will be required on the agent side:
+
+* *Go Agent*: certificate pinning through {apm-go-ref}/configuration.html#config-server-cert[`ELASTIC_APM_SERVER_CERT`]
+* *Python Agent*: certificate pinning through {apm-py-ref}/configuration.html#config-server-cert[`server_cert`]
+* *Ruby Agent*: certificate pinning through {apm-ruby-ref}/configuration.html#config-ssl-ca-cert[`server_ca_cert`]
```

Comment:
@elastic/apm-agent-ruby is this config about certificate pinning? If so, it should be aligned with the Python and Go agents - `ELASTIC_APM_SERVER_CERT`. Otherwise, if it is about setting a custom CA, it should be aligned with Node JS agent's `ELASTIC_APM_SERVER_CA_CERT_FILE` (not suggesting which should adjust 🙂 )

Reply:
It's the latter, and there's issues 😊 
Pinning: https://github.com/elastic/apm-agent-ruby/issues/248
CA cert file: https://github.com/elastic/apm-agent-ruby/issues/650